Sources/BSON/ExtendedJSONEncoder.swift

```diff
@@ -0,0 +1,45 @@
+import Foundation
+
+/// Facilitates the encoding of `Encodable` values into ExtendedJSON.
+public class ExtendedJSONEncoder {
+    /// An enum representing one of the two supported string formats based on the JSON standard
+    /// that describe how to represent BSON documents in JSON using standard JSON types and/or type wrapper objects.
+    public enum Mode {
+        /// Canonical Extended JSON Format: Emphasizes type preservation
+        /// at the expense of readability and interoperability.
+        case canonical
+
+        /// Relaxed Extended JSON Format: Emphasizes readability and interoperability
+        /// at the expense of type preservation.
+        case relaxed
+    }
+
+    /// Determines whether to encode to canonical or relaxed extended JSON. Default is relaxed.
+    public var mode: Mode = .relaxed
+
+    /// Contextual user-provided information for use during encoding.
+    public var userInfo: [CodingUserInfoKey: Any] = [:]
+
+    /// Initialize an `ExtendedJSONEncoder`.
+    public init() {
+        fatalError("unimplemented")
+    }
+
+    /// Encodes an instance of the Encodable Type `T` into Data representing canonical or relaxed extended JSON.
+    /// The value of `mode` will determine which format is used. If it is not set explicitly, relaxed will be used.
```

Comment:
maybe say `self.mode` just to be clear it's a property and not supposed to be an option to this method or something

Reply:
ok